src/main/java/net/sf/jabref/gui/importer/actions/OpenDatabaseAction.java

```diff
@@ -171,111 +173,71 @@ private void openTheFile(File file, boolean raisePanel) {
-                            Localization.lang("Error"), JOptionPane.ERROR_MESSAGE);
+            }
 
-                    String message = "<html>" + errorMessage + "<p>"
-                            + (tryingAutosave ? Localization.lang(
-                                    "Error opening autosave of '%0'. Trying to load '%0' instead.",
-                                    file.getName()) : ""/*Globals.lang("Error opening file '%0'.", file.getName())*/)
-                            + "</html>";
-                    JOptionPane.showMessageDialog(null, message, Localization.lang("Error opening file"),
-                            JOptionPane.ERROR_MESSAGE);
-
-                    if (tryingAutosave) {
-                        tryingAutosave = false;
-                        fileToLoad = file;
-                    } else {
-                        done = true;
-                    }
-                    continue;
-                } else {
-                    done = true;
-                }
+            if (BackupManager.checkForBackupFile(fileToLoad.toPath())) {
+                BackupUIManager.showRestoreBackupDialog(frame, fileToLoad.toPath());
+            }
 
-                final BasePanel panel = addNewDatabase(result, file, raisePanel);
-                if (tryingAutosave) {
-                    panel.markNonUndoableBaseChanged();
-                }
+            ParserResult result;
+            try {
+                result = OpenDatabase.loadDatabase(fileToLoad, Globals.prefs.getImportFormatPreferences());
+            } catch (IOException ex) {
+                LOGGER.error("Error loading database " + fileToLoad, ex);
+                result = ParserResult.getNullResult();
+                JOptionPane.showMessageDialog(null, Localization.lang("Error opening file") + " '" + fileName + "'",
+                        Localization.lang("Error"), JOptionPane.ERROR_MESSAGE);
+            }
 
-                // After adding the database, go through our list and see if
-                // any post open actions need to be done. For instance, checking
-                // if we found new entry types that can be imported, or checking
-                // if the database contents should be modified due to new features
-                // in this version of JabRef:
-                final ParserResult finalReferenceToResult = result;
-                SwingUtilities.invokeLater(
-                        () -> OpenDatabaseAction.performPostOpenActions(panel, finalReferenceToResult, true));
+            if (Objects.nonNull(result.getDatabase().getDatabaseID())) {
+                try {
+                    new SharedDatabaseUIManager(frame).openSharedDatabaseFromParserResult(result);
+                } catch (SQLException | DatabaseNotSupportedException | InvalidDBMSConnectionPropertiesException e) {
+                    result.getDatabaseContext().setDatabaseFile(null); // do not open the original file
```

Comment:
I think our convention for setters was that the never except a null value but there is a clear* method for this case.